### ISSUE TITLE
Use `flutter test` instead of `flutter drive`

### ISSUE THREAD

**Issue by mateuszwojtczak** [leancodepl/patrol#421]

**Comment by bartekpacia** [comment]
`flutter test` currently doesn't allow for using custom bindings. See https://github.com/flutter/flutter/issues/113097.

We have our own custom binding, `PatrolBinding`. The only things it does is setting the source of every pointer event (tap, slide, etc) to `test` instead of `device`.

It lets native automators tap on our Flutter app (the most common use case for this is a WebView).

Without our custom bindings, when we `$.native.tap()` on some widget, no tap is performed, instead we only get the following output in the console:

```
Some possible finders for the widgets at Offset(195.0, 779.0):
  find.text('Open webview screen')
  find.widgetWithText(Align, 'Open webview screen')
  find.widgetWithText(Padding, 'Open webview screen')
  find.widgetWithText(GestureDetector, 'Open webview screen')
  find.widgetWithText(MouseRegion, 'Open webview screen')
  find.widgetWithText(InkWell, 'Open webview screen')
  find.byType(SliverList)
  find.byType(SliverPadding)
  find.byType(Viewport)
  find.byType(Scrollable)
  find.byKey(const Key('listViewKey'))
  find.byType(NotificationListener<ScrollNotification>)
  find.byKey(const Key('scaffold'))
  find.byElementType(InheritedModelElement<Object>)
```

**Comment by bartekpacia** [comment]
Replacing `flutter drive` with `flutter test` should also fix #425 ([source](https://github.com/flutter/flutter/issues/71594)).

**Comment by bartekpacia** [comment]
We won't need `PatrolBinding` at all when https://github.com/flutter/flutter/pull/108430 lands on the `stable` channel.

**Comment by bartekpacia** [comment]
```mermaid
graph TD
  fluttertest[flutter test + JUnit #421, #495] --> flutter3.5
  host_features[screenshots, scripts] --> host_server[vmservice #593]
  host_server --> custom_bindings[requires PatrolBinding] --> blocked_by_flutter[blocked by flutter #113097]
```

**Comment by bartekpacia** [comment]
This is superseded by #661.
